docs/notebooks/qhsri-tutorial.pct.py

```diff
@@ -0,0 +1,232 @@
+model = GaussianProcessRegression(gpflow_model)
+
+# %% [markdown]
+# Now consider how we might want to select a batch of $q$ query points to observe. It would be useful for some of these to be "safe bets" that we think are very likely to provide an improvement. It would also be valuable for some of these to be sampling parts of the space that we have no idea what the expected observed value is (i.e. high variance). This problem is very similar to that encountered in building finance portfolios, where you want a mix of high  risk/high reward and low risk/low reward assets. You would also want to know how much of your total capital to invest in each asset.
+#
+# We can sample from the model to determine the mean and standard deviation of points. For example if we randomly sample 1000 points from the space, we can then plot the mean and standard deviation. As we actually want to maximise standard deviation, we will multiply by -1 once sampled
+
+# %%
+uniform_points = tf.random.uniform([1000, 2], dtype="float64")
+uniform_pts_mean, uniform_pts_var = model.predict(uniform_points)
+uniform_pts_std = -tf.sqrt(uniform_pts_var)
+
+plt.scatter(uniform_points[:, 0], uniform_points[:, 1])
+plt.title("Uniformly sampled points")
+plt.show()
+plt.close()
+
+plt.scatter(uniform_pts_mean, uniform_pts_std)
+plt.title("Mean vs negative std of uniformly sampled points")
+plt.xlabel("Mean")
+plt.ylabel("Negative std")
+plt.show()
+
+# %% [markdown]
+# Since we only want the best points in terms of the risk reward tradeoff, we want points that have low means and high standard deviations. So we remove any points that are dominated by another point. A point `a` is dominated by another point `b` if `b.mean` <= `a.mean` and `b.std` >= `a.std`.
+#
+# There is a function in trieste that lets us calculate this non-dominated set. Let's find the non-dominated points and plot them on the above chart.
+
+# %%
+from trieste.acquisition.multi_objective.dominance import non_dominated
+
+uniform_non_dominated = non_dominated(tf.concat([uniform_pts_mean, uniform_pts_std], axis=1))[0]
+
+plt.scatter(uniform_pts_mean, uniform_pts_std)
+plt.scatter(uniform_non_dominated[:, 0], uniform_non_dominated[:, 1], c="r")
+plt.title("Mean vs negative std of uniformly sampled points")
+plt.xlabel("Mean")
+plt.ylabel("Negative std")
+plt.show()
+
+print(f"There are {len(uniform_non_dominated)} non-dominated points")
+
+# %% [markdown]
+# We can see that there's only a few non-dominated points to choose from the select the next batch. This set of non-dominated points is the Pareto front in the optimisation task of minimising mean and maximising standard deviation.
+#
+# This means we can improve on this by using optimisation rather than random sampling to pick the points that we will select our batch from. In trieste we use the NSGA-II multi-objective optimisation method from the [pymoo](https://pymoo.org/) library to find the Pareto front.
```

Comment:
Mention (either here or elsewhere) how to install the optional qhsri dependency?